packages/xo-web/src/xo-app/vm/tab-general.js

```diff
@@ -194,7 +194,15 @@ const GeneralTab = decorate([
                 <Icon icon='memory' size='lg' />
               </span>
             </h2>
-            <BlockLink to={`/vms/${id}/stats`}>{statsOverview && <MemorySparkLines data={statsOverview} />}</BlockLink>
+            {vm.managementAgentDetected === true && vm.managementAgentDetected !== undefined ? (
+              <BlockLink to={`/vms/${id}/stats`}>
+                {statsOverview && <MemorySparkLines data={statsOverview} />}
+              </BlockLink>
+            ) : (
+              <BlockLink to={`/vms/${id}/stats`}>
+                <Icon color='text-warning' icon='alarm' /> {_('guestToolsNecessary')}
```

Comment:
Let's keep it consistent with the other warning and make the whole label yellow:
```suggestion
                <span className='text-warning'><Icon icon='alarm' /> {_('guestToolsNecessary')}</span>
```